teuthology/orchestra/remote.py

```diff
@@ -401,6 +401,17 @@ def get_file(self, path, sudo=False, dest_dir='/tmp'):
             self.remove(path)
         return local_path
 
+    def write_file(self, path, data):
+        """
+        Write a file on a remote machine.
+
+        :param path: Path on the remote being written to.
+        :param data: Data to be written.
+        """
+        tempfilename = tempfile.mkstemp()[1]
+        self.run(['tee', tempfilename], stdin=data)
```

Comment:
If you are teeing the data on a remote why not to do it to the temporary destination, it is better, to do using `path` directly in the first place and by the way it is easy to add sudo support here as well, like:
```
. . .  (path, data, sudo=False):
  tee = 'sudo tee' if sudo else 'tee'
  self.run(args='%s %s' % (tee, path), stdin=data)
```

Reply:
> If you are teeing the data on a remote why not to do it to the temporary destination, it is better, to do using path directly in the first place

`tempfilename` already holds a path for temporary destination.

Reply:
> and by the way it is easy to add sudo support here as well, like:

Done.